src/node/build/index.ts

```diff
@@ -214,7 +215,7 @@ export async function build(options: BuildConfig = {}): Promise<BuildResult> {
       createReplacePlugin(
         {
           'process.env.NODE_ENV': '"production"',
-          'process.env.': `({}).`,
+          'process.env.': `(${JSON.stringify(env)}).`,
```

Comment:
Wouldn't it be better to write this instead? (Note the removal of the trailing dot `.` and the addition of `NODE_ENV` to the process env object.)
```js
'process.env': `(${JSON.stringify({ ...env, NODE_ENV: 'production'})})`
```

Then, cases such as `process.env["NODE_ENV"]` would also get handled.